sasl/mechanism.cpp

```diff
@@ -389,28 +422,30 @@ void SaslMechanism::verify()
             setState( Succeeded );
         else
             setState( Failed );
+        return;
     }
-    else if ( d->ldapRelay ) {
+
+    if ( storedSecret().isEmpty() || storedSecret() == secret() ) {
```

Comment:
this was preserved from the original code, but I do not understand why you'd want to return `Succeeded` if storedSecret() is empty?